src/lsp/cobol_lsp/lsp_request.ml

```diff
@@ -454,60 +454,48 @@ let handle_semtoks_full,
-  let is_not_in_hovered_def = not @@ Lsp_position.is_in_srcloc ~filename pos def_loc in
-  if is_not_in_hovered_def || always_show_hover_text_in_data_div
-    then text
-    else raise Not_found
-
-
-let get_hover_text_and_loc cu_name element_at_pos position group filename =
+let get_hover_text_and_defloc (qn: Cobol_ptree.qualname) (cu: Cobol_unit.Types.cobol_unit) =
+  let data_def =
+    try Cobol_unit.Qualmap.find qn cu.unit_data.data_items.named
+    with _ -> raise Not_found
+  in
+  let hover_text = Pretty.to_string "%a" Lsp_data_info_printer.pp_data_definition data_def in
+  let loc =
+    match data_def with
+      | Data_field { def; _} -> ~@def
+      | Data_renaming { def; _} -> ~@def
+      | Data_condition { def; _} -> ~@def
+      | Table_index { table; _ } -> ~@table
+  in hover_text, loc
+
+let get_hover_info cu_name element_at_pos group =
   let { payload = cu; _ } = CUs.find_by_name cu_name group in
-  match element_at_pos with
-    | Data_item { full_qn = Some qn; def_loc } ->
-        let text,_ = get_hover_text qn cu in
-        filter_data_div text def_loc position filename, Some def_loc
-    | Data_full_name qn ->
-        let text,loc = get_hover_text qn cu in
-          Option.fold ~none:text ~some:(fun loc -> filter_data_div text loc position filename) loc,
-          Some (Lsp_lookup.baseloc_of_qualname qn)
-    | Data_name qn ->
-        let text,_ = get_hover_text qn cu in
-         text, Some (Lsp_lookup.baseloc_of_qualname qn)
-    | Data_item _ | Proc_name _ ->
-        raise Not_found
+    match element_at_pos with
+      | Data_item { full_qn = Some qn; def_loc } ->
+          let text, _ = get_hover_text_and_defloc qn cu in
+          text, def_loc, def_loc
+      | Data_full_name qn | Data_name qn ->
+          let text, def_loc = get_hover_text_and_defloc qn cu in
+          text, def_loc, Lsp_lookup.baseloc_of_qualname qn
+      | Data_item _ | Proc_name _ ->
+          raise Not_found
 
 let lookup_hover_definition_in_doc
     HoverParams.{ textDocument = doc; position; _ }
     Cobol_typeck.Outputs.{ group; _ } =
       let filename = Lsp.Uri.to_path doc.uri in
       match Lsp_lookup.element_at_position ~uri:doc.uri position group with
-  | { element_at_position = None; _ }
-  | { enclosing_compilation_unit_name = None; _ } ->
-      raise Not_found
-  | { element_at_position = Some ele_at_pos;
-      enclosing_compilation_unit_name = Some cu_name } ->
-        get_hover_text_and_loc cu_name ele_at_pos position group filename
+        | { element_at_position = None; _ }
+        | { enclosing_compilation_unit_name = None; _ } ->
+            raise Not_found
+        | { element_at_position = Some ele_at_pos;
+          enclosing_compilation_unit_name = Some cu_name } ->
+            let hover_text, def_loc, hover_loc = get_hover_info cu_name ele_at_pos group in
+            if always_show_hover_text_in_data_div || not (Lsp_position.is_in_srcloc ~filename position def_loc)
```

Comment:
Please cut the line after the `||`.
In general, it's best to avoid lines that exceed 80 characters.